Dockerfile

```diff
@@ -0,0 +1,50 @@
+FROM ruby:alpine
+MAINTAINER "Unif.io, Inc. <support@unif.io>"
+
+# This is the release of https://github.com/hashicorp/docker-base to pull in order
+# to provide HashiCorp-built versions of basic utilities like dumb-init and gosu.
+ENV DOCKER_BASE_VERSION=0.0.4
+
+ARG GEMFURY_SOURCE_URL_TOKEN
+
+RUN mkdir -p /usr/local/external_bins && \
+
+    # docker base goodies
+    apk add --no-cache --update ca-certificates gnupg openssl git wget unzip && \
+    gpg --recv-keys 91A6E7F85D05C65630BEF18951852D87348FFC4C && \
+    mkdir -p /tmp/build && \
+    cd /tmp/build && \
+    wget -q "https://releases.hashicorp.com/docker-base/${DOCKER_BASE_VERSION}/docker-base_${DOCKER_BASE_VERSION}_linux_amd64.zip" && \
+    wget -q "https://releases.hashicorp.com/docker-base/${DOCKER_BASE_VERSION}/docker-base_${DOCKER_BASE_VERSION}_SHA256SUMS" && \
+    wget -q "https://releases.hashicorp.com/docker-base/${DOCKER_BASE_VERSION}/docker-base_${DOCKER_BASE_VERSION}_SHA256SUMS.sig" && \
+    gpg --batch --verify docker-base_${DOCKER_BASE_VERSION}_SHA256SUMS.sig docker-base_${DOCKER_BASE_VERSION}_SHA256SUMS && \
+    grep docker-base_${DOCKER_BASE_VERSION}_linux_amd64.zip docker-base_${DOCKER_BASE_VERSION}_SHA256SUMS | sha256sum -c && \
+    unzip docker-base_${DOCKER_BASE_VERSION}_linux_amd64.zip && \
+    cp bin/gosu /usr/local/bin && \
+    cp bin/dumb-init /usr/local/bin && \
+
+    # Covalence
+    apk add --no-cache --update build-base && \
+    gem install covalence --source "https://${GEMFURY_SOURCE_URL_TOKEN}@gem.fury.io/me/" --no-ri --no-rdoc && \
+    gem install dotenv serverspec --no-ri --no-rdoc && \
+    gem install rspec ci_reporter_rspec --no-ri --no-rdoc
+
+ENV DOCKER_VERSION 1.9.1
+ENV DOCKER_SHA256 6a095ccfd095b1283420563bd315263fa40015f1cee265de023efef144c7e52d
+
+RUN wget -q -O docker.tgz "https://get.docker.com/builds/Linux/x86_64/docker-${DOCKER_VERSION}.tgz" && \
```

Comment:
I am liking the split out sections for the long run command, nice that it allows comments and whitespace in the middle.